scripts/update_sales_values.ts

```diff
@@ -0,0 +1,187 @@
+import { getAddress } from "viem";
+import { getDeployment } from "../src/utils/getDeployment.js";
+import { getHypercertTokenId } from "../src/utils/tokenIds.js";
+import { HypercertExchangeAbi } from "@hypercerts-org/sdk";
+import { EvmClientFactory } from "../src/clients/evmClient.js";
+
+const getChain = (chainId: number) => {
+  const chains: Record<number, Chain> = {
+    10: optimism,
+    314: filecoin,
+    8453: base,
+    42161: arbitrum,
+    42220: celo,
+    84532: baseSepolia,
+    314159: filecoinCalibration,
+    421614: arbitrumSepolia,
+    11155111: sepolia,
+  };
+
+  const chain = chains[chainId];
+  if (!chain) throw new Error(`Unsupported chain ID: ${chainId}`);
+  return chain;
+};
+
+const main = async () => {
+  console.log("update_sales_values");
+  // Get all sales rows
+  // Create supabase client
+  const supabase = createClient(
+    process.env.SUPABASE_CACHING_DB_URL!,
+    process.env.SUPABASE_CACHING_SERVICE_API_KEY!,
+  );
+  const salesResponse = await supabase.from("sales").select("*");
+  const sales = salesResponse.data;
+
+  if (!sales) {
+    console.log("No sales found");
+    return;
+  }
+
+  const results: TakerBid[] = [];
+  for (const sale of sales) {
+    let chainId: number | undefined;
+    const tokenId = BigInt(sale.item_ids[0]);
+    const claimId = getHypercertTokenId(tokenId);
+    const hypercert_id = sale.hypercert_id.replace("undefined", claimId);
+
+    try {
+      chainId = parseClaimOrFractionId(hypercert_id).chainId;
+      if (!chainId) {
+        console.log("No chainId found for sale", sale);
+        continue;
+      }
+    } catch (e) {
+      console.log("Error parsing hypercert_id", hypercert_id);
+      continue;
+    }
+    // Get transaction and parse logs using viem
+
+    const client = EvmClientFactory.createClient(Number(chainId));
+    const { addresses } = getDeployment(Number(chainId));
+    const { currency } = sale;
+
+    try {
+      const transactionReceipt = await client.getTransactionReceipt({
+        hash: sale.transaction_hash as `0x${string}`,
+      });
+
+      // parse logs to get claimID, contractAddress and cid
+      const transactionLogsHypercertMinter = transactionReceipt.logs.filter(
+        (log) =>
+          log.address.toLowerCase() ===
+          addresses?.HypercertMinterUUPS?.toLowerCase(),
+      );
+
+      let currencyAmount = 0n;
+      if (currency === zeroAddress) {
+        // Get value of the transaction
+        const transaction = await client.getTransaction({
+          hash: sale.transaction_hash as `0x${string}`,
+        });
+        currencyAmount = transaction.value;
+      } else {
+        const currencyLogs = transactionReceipt.logs.filter(
+          (log) => log.address.toLowerCase() === currency.toLowerCase(),
+        );
+        const parsedCurrencyLogs = parseEventLogs({
+          abi: erc20Abi,
+          logs: currencyLogs,
+        });
+        const transferLog = parsedCurrencyLogs.find(
```

Comment:
There are actually multiple transfer events because a part is routed as the protocolFee: https://sepolia.etherscan.io/tx/0xda244abd0239a4ee149f33cfc469d597290af7cfa55f438902aacbbd8f934b1f#eventlog

So, `Array.find` will only return part of the total value transacted